dbclient/HiveClient.py

```diff
@@ -426,21 +440,26 @@ def log_all_tables(self, db_name, cid, ec_id, metastore_dir, err_log_path, succe
         batch_size = 100    # batch size to iterate over databases
         num_of_buckets = (num_of_tables // batch_size) + 1     # number of slices of the list to take
 
-        all_tables = []
         with open(success_log_path, 'a') as sfp:
             for m in range(0, num_of_buckets):
                 tables_slice = 'print(all_tables[{0}:{1}])'.format(batch_size*m, batch_size*(m+1))
                 results = self.submit_command(cid, ec_id, tables_slice)
                 table_names = ast.literal_eval(results['data'])
                 for table_name in table_names:
                     print("Table: {0}".format(table_name))
-                    is_successful = self.log_table_ddl(cid, ec_id, db_name, table_name, metastore_dir,
-                                                       err_log_path, has_unicode)
-                    if is_successful == 0:
-                        print(f"Exported {db_name}.{table_name}")
-                        success_item = {'table': f'{db_name}.{table_name}', 'iam': iam}
+                    full_table_name = f'{db_name}.{table_name}'
+                    if checkpoint_metastore_set.contains(full_table_name):
```

Comment:
I think this usage of checkpoint set is correct. But it's easily to be broken in future changes. Could you leave comments to explain why checkpointing is needed, how it works and the potential risks (e.g. why flush is necessary)? 

I know it's over-documenting, but I don't think the code is future proof in the current situation.